datacube/virtual/impl.py

```diff
@@ -525,6 +525,10 @@ def group(self, datasets: VirtualDatasetBag, **group_settings: Dict[str, Any]) -
         dim = self.get('dim', 'time')
 
         def to_box(value):
+            # xarray.DataArray.groupby weirdness: sometimes the dim is dropped for no apparent reasons
+            if dim not in value.dims:
```

Comment:
maybe you should use `squeeze=False` parameter of `groupby` rather than having this?

Reply:
That worked! Thanks @Kirill888.